block/state.go

```diff
@@ -32,6 +32,10 @@ func (m *Manager) LoadStateOnInit(store store.Store, genesis *tmtypes.GenesisDoc
 	}
 
 	m.State = s
+
+	if m.State.LastGossipedHeight == 0 {
```

Comment:
worth adding a comment here explainig the reasoning

Reply:
comment added